### PR TITLE
Add total holdings.

### PR DESCRIPTION
**Pull request by jurgen178**
Add the total holdings with total gain/loss to the header in the portfolio fragment.

### CODE REVIEW

**Comment by nassemz** [comment]
great job.  just there is a problem when you have a lot of holdings in your portfolio , as you present the total and then each one and if you have many it will look bad, i think total is enough.

**Comment by jurgen178** [comment]
Thanks naessemz,
the header contains the total holdings, followed by the total gains and total losses in the next line.

**Comment by nassemz** [comment]
try to have 10 ~15 holdings in your profile and send screenshot

**Comment by jurgen178** [comment]
Screenshot_20200525-125329

**Comment by premnirmal** [comment]
Hi @jurgen178

Thank you for the changes. I have some thoughts on the total holdings UI, I will post the review this weekend. 

But first:

Can you keep the after hours price change in a separate PR?

In case I need to revert one of the changes, that will not be possible if both changes are bundled together in one PR.

It also makes it easier to review if the changes are specific and in separate PRs

Thanks

**Comment by jurgen178** [comment]
Hi @premnirmal I have changed the PR back to the previous commit.

**Comment by Kapaka7** [comment]
@premniral, could you also add a separate 1x1 widget, only with the sum of the entire portfolio.  as in the picture 
IMG_20200530_195831

File: app/src/main/kotlin/com/github/premnirmal/ticker/model/IStocksProvider.kt [hunk 1]
```diff
@@ -55,4 +55,8 @@ interface IStocksProvider {
     ticker: String,
     holding: Holding
   )
+
+  fun totalHoldings(): Pair<String, Int>
+
+  fun totalGainLoss(): Pair<String, String>
```
Comment (premnirmal) [inline app/src/main/kotlin/com/github/premnirmal/ticker/model/IStocksProvider.kt:61]: Move this logic to a viewmodel and use the VM inside `HomeFragment`.

Prefer not to have calculation logic like this in the model here, instead if should be part of the UI binding code.

Comment (jurgen178) [reply app/src/main/kotlin/com/github/premnirmal/ticker/model/IStocksProvider.kt:61]: Thanks @premnirmal for your feedback. My first thought for the totals were to use a dedicated stock quote block that displays the totals instead of the individual stock data, but this would require quite a large change in the design. Having the totals in the header looked as the perfect place when I created a mockup. Courious to see what else could be used!

The StocksProvider seemed like a perfect place for the totals calculation. Will check out the VM.

**Comment by jurgen178** [comment]
Thanks @premnirmal for your feedback. My first thought for the totals were to use a dedicated stock quote block that displays the totals instead of the individual stock data, but this would require quite a large change in the design. Having the totals in the header looked as the perfect place when I created a mockup. Curious to see what else could be used!

**Comment by jurgen178** [comment]
Build failed with
  Execution failed for task ':app:countPurefossDebugDexMethods'.
  > The current APK has 70137 methods, the current max is: 64000.

It is probably caused by the updated settings to the latest versions and the new ViewModel as the recent commit is rather small. Haven't seen this error before, so I'm not sure what the best fix would be.

File: app/build.gradle [hunk 2]
```diff
@@ -148,7 +148,7 @@ android {
       debuggable true
       versionNameSuffix "-DEBUG"
       ext.enableCrashlytics = false
-      minifyEnabled true
+      minifyEnabled false
```
Comment (premnirmal) [inline app/build.gradle:151]: is this change needed?

File: app/build.gradle [hunk 3]
```diff
@@ -170,6 +170,13 @@ android {
     exclude "META-INF/LICENSE.txt"
     exclude "META-INF/NOTICE.txt"
   }
+    
+  lintOptions {
+        checkReleaseBuilds false
+        // Or, if you prefer, you can continue to check for errors in release builds,
+        // but continue the build even when errors are found:
+        abortOnError false
```
Comment (premnirmal) [inline app/build.gradle:178]: why was this added?

File: app/build.gradle [hunk 4]
```diff
@@ -204,7 +211,8 @@ dependencies {
   implementation "javax.annotation:javax.annotation-api:1.3.2"
 
   implementation "com.google.dagger:dagger:$DAGGER_VERSION"
-  kapt "com.google.dagger:dagger-compiler:$DAGGER_VERSION"
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.2.0'
```
Comment (premnirmal) [inline app/build.gradle:214]: what does this dependency provide?

File: app/src/main/kotlin/com/github/premnirmal/ticker/home/TotalHoldingsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package com.github.premnirmal.ticker.home
+
+import androidx.lifecycle.ViewModel
+import com.github.premnirmal.ticker.model.IStocksProvider
+import com.github.premnirmal.ticker.network.data.Quote
+
+class TotalHoldingsViewModel(private val stocksProvider: IStocksProvider): ViewModel() {
```
[the remark this answers is not in the document]
Comment (jurgen178) [reply app/src/main/kotlin/com/github/premnirmal/ticker/home/TotalHoldingsViewModel.kt]: Fixed.

File: app/build.gradle (hunk 2 above)
Comment (jurgen178) [reply app/build.gradle:151]: Without this, I cannot step through the code.

File: app/build.gradle (hunk 3 above)
Comment (jurgen178) [reply app/build.gradle:178]: Got an build error and this was suggested in the error message to add to the gradle setting.
I'm using Android Studio 4.0 on Windows, all new installation.

File: app/build.gradle (hunk 2 above)
Comment (premnirmal) [reply app/build.gradle:151]: The buildApk task fails due to this, because it doesn't minify the debug build

**Comment by panachoi** [comment]
What would also be nice would be to have the individual holdings value in the widget (also maybe total/daily gain/loss).
